modules/cache-common/src/main/java/org/opensearch/cache/common/tier/TieredSpilloverCache.java

```diff
@@ -69,254 +72,529 @@ public class TieredSpilloverCache<K, V> implements ICache<K, V> {
     private static final List<RemovalReason> SPILLOVER_REMOVAL_REASONS = List.of(RemovalReason.EVICTED, RemovalReason.CAPACITY);
     private static final Logger logger = LogManager.getLogger(TieredSpilloverCache.class);
 
-    private final ICache<K, V> diskCache;
-    private final ICache<K, V> onHeapCache;
-
-    // Removal listeners for the individual tiers
-    private final RemovalListener<ICacheKey<K>, V> onDiskRemovalListener;
-    private final RemovalListener<ICacheKey<K>, V> onHeapRemovalListener;
-
-    // Removal listener from the spillover cache as a whole
-    private final RemovalListener<ICacheKey<K>, V> removalListener;
+    static final String NUMBER_OF_SEGMENTS_ZERO_EXCEPTION_MESSAGE = "Number of segments cannot be less than or equal " + "to zero";
 
     // In future we want to just read the stats from the individual tiers' statsHolder objects, but this isn't
     // possible right now because of the way computeIfAbsent is implemented.
     private final TieredSpilloverCacheStatsHolder statsHolder;
     private ToLongBiFunction<ICacheKey<K>, V> weigher;
     private final List<String> dimensionNames;
-    ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
-    ReleasableLock readLock = new ReleasableLock(readWriteLock.readLock());
-    ReleasableLock writeLock = new ReleasableLock(readWriteLock.writeLock());
-    /**
-     * Maintains caching tiers in ascending order of cache latency.
-     */
-    private final Map<ICache<K, V>, TierInfo> caches;
+
     private final List<Predicate<V>> policies;
```

Comment:
nit: Do we still need the policies here?

Reply:
Not needed, will remove.